tests/foreman/api/test_contentmanagement.py

```diff
@@ -932,6 +934,120 @@ def test_positive_update_with_immediate_sync(self):
+            'search': 'name={} and version={}'
+                      .format(module_name, module_versions[0])
+        })[0]
+        # Add puppet module to the CV
+        entities.ContentViewPuppetModule(
+            content_view=content_view,
+            id=puppet_module_old.id,
+        ).create()
+        content_view = content_view.read()
+        self.assertGreater(len(content_view.puppet_module), 0)
+        # Publish and promote CVV
+        content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 1)
+        cvv = content_view.version[-1].read()
+        promote(cvv, lce.id)
+        cvv = cvv.read()
+        self.assertEqual(len(cvv.environment), 2)
+        # Wait till capsule sync finishes
+        sync_status = capsule.content_get_sync()
+        self.assertTrue(
+            len(sync_status['active_sync_tasks']) >= 1 or
+            sync_status['last_sync_time']
+        )
+        for task in sync_status['active_sync_tasks']:
+            entities.ForemanTask(id=task['id']).poll()
+        sync_status = capsule.content_get_sync()
+        last_sync_time = sync_status['last_sync_time']
+        sync_errors_before = sync_status['last_failed_sync_tasks']
+        # Unassign old puppet module version from CV
+        entities.ContentViewPuppetModule(
+            content_view=content_view,
+            id=content_view.puppet_module[0].id,
+        ).delete()
+        # Assign new puppet module version
+        puppet_module_new = entities.PuppetModule().search(query={
+            'search': 'name={} and version={}'
+                      .format(module_name, module_versions[1])
+        })[0]
+        entities.ContentViewPuppetModule(
+            content_view=content_view,
+            id=puppet_module_new.id,
+        ).create()
+        self.assertGreater(len(content_view.puppet_module), 0)
+        # Publish and promote CVV
+        content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 2)
+        cvv = content_view.version[-1].read()
+        promote(cvv, lce.id)
+        cvv = cvv.read()
+        self.assertEqual(len(cvv.environment), 2)
+        # Wait till capsule sync finishes
+        sync_status = capsule.content_get_sync()
+        if sync_status['active_sync_tasks']:
+            for task in sync_status['active_sync_tasks']:
+                entities.ForemanTask(id=task['id']).poll()
+        else:
+            self.assertNotEqual(
+                sync_status['last_sync_time'], last_sync_time)
+        # Polling the task will raise an assertion in case of error, but in
+        # case we couldn't catch tasks running in time, let's ensure no new
+        # failed tasks were logged for capsule
+        sync_status = capsule.content_get_sync()
+        sync_errors_after = sync_status['last_failed_sync_tasks']
+        self.assertEqual(len(sync_errors_before), len(sync_errors_after))
```

Comment:
Could we check that actually last version of module present on capsule? Or what that sync procedure does?

Reply:
Yep, we can. BZ was about inability to perform sync successfully, and also we have multiple tests which are checking whether there's latest version of package available after sync, that why i skipped such check for this specific test. Want me to add it?

Reply:
Then honestly, I don't understand why do we need all this code at all:
```
+        # Wait till capsule sync finishes
 +        sync_status = capsule.content_get_sync()
 +        if sync_status['active_sync_tasks']:
 +            for task in sync_status['active_sync_tasks']:
 +                entities.ForemanTask(id=task['id']).poll()
 +        else:
 +            self.assertNotEqual(
 +                sync_status['last_sync_time'], last_sync_time)
 +        # Polling the task will raise an assertion in case of error, but in
 +        # case we couldn't catch tasks running in time, let's ensure no new
 +        # failed tasks were logged for capsule
 +        sync_status = capsule.content_get_sync()
 +        sync_errors_after = sync_status['last_failed_sync_tasks']
+         self.assertEqual(len(sync_errors_before), len(sync_errors_after))
```
just check that action was really did what is has to do